providers/dns/cloudflare/cloudflare.go

```diff
@@ -61,37 +67,25 @@ func NewDNSProvider() (*DNSProvider, error) {
 	return NewDNSProviderConfig(config)
 }
 
-// NewDNSProviderCredentials uses the supplied credentials
-// to return a DNSProvider instance configured for Cloudflare.
-// Deprecated
-func NewDNSProviderCredentials(email, key string) (*DNSProvider, error) {
-	config := NewDefaultConfig()
-	config.AuthEmail = email
-	config.AuthKey = key
-
-	return NewDNSProviderConfig(config)
-}
-
 // NewDNSProviderConfig return a DNSProvider instance configured for Cloudflare.
 func NewDNSProviderConfig(config *Config) (*DNSProvider, error) {
 	if config == nil {
 		return nil, errors.New("cloudflare: the configuration of the DNS provider is nil")
 	}
 
-	client, err := NewClient(config.AuthEmail, config.AuthKey)
+	if config.TTL < minTTL {
+		config.TTL = minTTL
+	}
+
+	client, err := cloudflare.New(config.AuthKey, config.AuthEmail, cloudflare.HTTPClient(config.HTTPClient))
 	if err != nil {
 		return nil, err
 	}
 
-	client.HTTPClient = config.HTTPClient
-
 	// TODO: must be remove. keep only for compatibility reason.
 	client.BaseURL = CloudFlareAPIURL
```

Comment:
As we're already breaking the API (by removing the `NewDNSProviderCredentials`, could we possibly remove these as well?

Reply:
oops, I will restore `NewDNSProviderCredentials`